src/App.tsx

```diff
@@ -1,24 +1,169 @@
+  };
+
+  const addTodo = (title: string) => {
+    setInputDisabled(true);
+    const preparedTitle = title.trim();
+
+    if (!preparedTitle) {
+      setError(ErrorSpec.EMPTY_TITLE);
+      setHasErrors(true);
+      setInputDisabled(false);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    TodoService.createTodo(newTodo)
+      .then((responce) => {
+        setTempTodo(responce);
+        setTodosFromServer([...todosFromServer, responce]);
+      })
+      .catch(() => {
+        setError(ErrorSpec.NOT_ADDED);
+        setInputDisabled(false);
+        throw new Error();
+      })
+      .finally(() => {
+        setInputDisabled(false);
+        setTempTodo(null);
+      });
+  };
+
+  const removeTodo = (id: number) => {
+    setInputDisabled(true);
+    setIsLoading(filteredTodos.find((todo) => todo.id === id) || null);
+
+    TodoService.deleteTodo(id)
+      .then(() => setIsLoading(null))
+      .catch(() => {
+        setError(ErrorSpec.NOT_DELETED);
+        setInputDisabled(false);
+        throw new Error();
+      })
+      .finally(() => {
+        setTodosFromServer((prev) => prev.filter((todo) => todo.id !== id));
+        setInputDisabled(false);
+      });
+  };
+
+  const clearCompleted = () => {
+    filteredTodos.forEach(element => {
+      if (element.completed) {
+        removeTodo(element.id);
+      }
+    });
+    setUncompletedTodosCount(0);
+    setHasErrors(false);
+  };
+
+  useEffect(() => {
+    setIsLoading(
+      filteredTodos.find((todo) => todo.id === isLoading?.id) || null,
+    );
+  }, [filteredTodos, isLoading]);
+
   if (!USER_ID) {
+    setHasErrors(true);
+
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          onSubmit={addTodo}
+          onInput={cleanErrors}
+          inputDisabled={inputDisabled}
+          hasErrors={hasErrors}
+        />
+
+        {todosFromServer.length > 0 && (
+          <TodoList
+            todos={filteredTodos}
+            tempTodo={tempTodo}
+            deleteTodo={removeTodo}
+            isProcessing={isLoading}
+          />
+        )}
+        {todosFromServer.length > 0 && (
+          <Footer
+            onStatus={handleStatus}
+            status={status}
+            completedCount={uncompletedTodosCount}
+            handleClear={clearCompleted}
+            isClearNeeded={uncompletedTodosCount === filteredTodos.length}
+          />
+        )}
```

Comment:
The render condition is the same, so there's no need to split it into separate blocks